server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -712,19 +712,10 @@ public BytesRef parseBytesRef(Object value) {
                 if (v instanceof String s) {
                     builder.addString(f, s);
                 } else if (v instanceof Long || v instanceof Integer) {
-                    Long l = Long.valueOf(v.toString());
-                    // For a long encoded number, we must check if the number can be the encoded value
-                    // of an unsigned_long.
-                    Number ul = (Number) UNSIGNED_LONG_SHIFTED.format(l);
-                    if (l == ul) {
-                        builder.addLong(f, l);
-                    } else {
-                        long ll = UNSIGNED_LONG_SHIFTED.parseLong(String.valueOf(l), false, () -> 0L);
-                        builder.addUnsignedLong(f, ll);
-                    }
+                    builder.addLong(f, (Long) v);
```

Comment:
The test failures come because this is `Integer`. Could you break arm of the if statement up into `else if (v instanceof Long l)` and `else if (v instanceof Integer i)`? I think that'd fix it.